private/pkg/storage/storageos/bucket.go

```diff
@@ -284,14 +283,10 @@ func (b *bucket) validateExternalPath(path string, externalPath string) error {
 		// It's important that we detect these cases so that
 		// multi buckets don't unnecessarily fail when one of
 		// its delegates actually defines the path.
-		elements := strings.Split(normalpath.Normalize(externalPath), "/")
-		if len(elements) == 1 {
-			// The path is a single element, so there aren't
-			// any other files to check.
-			return err
-		}
-		for i := len(elements) - 1; i >= 0; i-- {
-			parentFileInfo, err := os.Stat(filepath.Join(elements[:i]...))
```

Comment:
According to [Join's doc](https://pkg.go.dev/path/filepath#Join), empty elements are ignored. This means if external path is `/foo/bar/baz`, it would check these paths here:

* `foo/bar`
* `foo`

but it should check:

* `/foo/bar`
* `/foo`